internal/controller/inference_pool.go

```diff
@@ -0,0 +1,409 @@
+		if isExtensionRefError {
+			resolvedRefsCondition := buildResolvedRefsCondition(inferencePool.Generation, "ai-gateway-controller", false, "ResolvedRefs", message)
+			conditions = append(conditions, resolvedRefsCondition)
+		} else {
+			// Add successful ResolvedRefs condition.
+			resolvedRefsCondition := buildResolvedRefsCondition(inferencePool.Generation, "ai-gateway-controller", true, "ResolvedRefs", "All references resolved successfully")
+			conditions = append(conditions, resolvedRefsCondition)
+		}
+
+		parents = append(parents, gwaiev1a2.PoolStatus{
+			GatewayRef: parentRef,
+			Conditions: conditions,
+		})
+	}
+
+	// If no Gateways reference this InferencePool, clear all parents.
+	// This correctly reflects that the InferencePool is not currently referenced by any Gateway.
+
+	inferencePool.Status.Parents = parents
+	if err := c.client.Status().Update(ctx, inferencePool); err != nil {
+		c.logger.Error(err, "failed to update InferencePool status")
+	}
+}
+
+// buildAcceptedCondition builds a condition for the InferencePool status.
+func buildAcceptedCondition(gen int64, controllerName string, conditionType string, message string) metav1.Condition {
+	status := metav1.ConditionTrue
+	reason := "Accepted"
+	if conditionType == "NotAccepted" {
+		status = metav1.ConditionFalse
+		reason = "NotAccepted"
+		conditionType = "Accepted"
+	}
+
+	return metav1.Condition{
+		Type:               conditionType,
+		Status:             status,
+		Reason:             reason,
+		Message:            fmt.Sprintf("InferencePool has been %s by controller %s: %s", reason, controllerName, message),
+		ObservedGeneration: gen,
+		LastTransitionTime: metav1.Now(),
+	}
+}
+
+// gatewayEventHandler returns an event handler for Gateway resources.
+func (c *InferencePoolController) gatewayEventHandler() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+		gateway, ok := obj.(*gwapiv1.Gateway)
+		if !ok {
+			return nil
+		}
+
+		// Find all InferencePools in the same namespace that might be affected by this Gateway.
+		var inferencePools gwaiev1a2.InferencePoolList
+		if err := c.client.List(ctx, &inferencePools, client.InNamespace(gateway.Namespace)); err != nil {
+			c.logger.Error(err, "failed to list InferencePools for Gateway event", "gateway", gateway.Name)
+			return nil
+		}
+
+		var requests []reconcile.Request
+		for _, pool := range inferencePools.Items {
+			// Check if this Gateway references the InferencePool.
+			if c.gatewayReferencesInferencePool(ctx, gateway, pool.Name, pool.Namespace) {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: client.ObjectKey{
+						Name:      pool.Name,
+						Namespace: pool.Namespace,
+					},
+				})
+			}
+		}
+
+		return requests
+	})
+}
+
+// routeEventHandler returns an event handler for AIGatewayRoute resources.
+func (c *InferencePoolController) routeEventHandler() handler.EventHandler {
```

Comment:
Let's be more explicit about the name to be consistent with httpRouteEventHandler below
```suggestion
// aiGatewayRouteEventHandler returns an event handler for AIGatewayRoute resources.
func (c *InferencePoolController) aiGatewayRouteEventHandler() handler.EventHandler {
```